spec/services/benefit_check_service_spec.rb

```diff
@@ -59,8 +59,16 @@
     context 'when the API raises an error' do
       let(:last_name) { 'SERVICEEXCEPTION' }
 
+      it 'raises a detailed error' do
+        expect { subject.call }.to raise_error(described_class::ApiError, /Service unavailable/)
+      end
+    end
+
+    context 'when the API times out' do
+      before { allow_any_instance_of(Savon::Client).to receive(:call).and_raise(Net::ReadTimeout) }
```

Comment:
Can you do this:
```ruby

before do
  savon_client = double(:savon_client)
  allow(savon_client).to receive(:call).and_raise(Net::ReadTimeout)
  allow(subject).to receive(:soap_client).and_return(savon_client)
end
```
Uglier, but avoids using `allow_any_instance_of` (if my code works). Not totally sure this is an improvement but worth suggesting.

Reply:
Yes it's a bit better